online_meetings/canvas_zoom_meetings.py

```diff
@@ -67,42 +72,61 @@ def get_zoom_json(self, page_num: int = 1) -> Dict:
             return zoom_json["result"]
         return None
 
+    def extract_from_js(self, key: str, text: str) -> Optional[str]:
+        """Takes a javascript text and attempts to extract a key/value (May not work with everything)
+        
+        :param key: key to extract
+        :type key: str
+        :param text: text to search in
+        :type text: str
+        :return: The matching string or None if not found
+        :rtype: Optional[str]
+        """
+        pattern = re.search(f'{key}.*"(.*)"', text)
+        if pattern:
+            return pattern.group(1)
+        return None
+
     def get_zoom_details(self, url: str, data: Dict[str, str], course_id: int):
         # Start up the zoom session
         # Initiate the LTI launch to Zoom in a session
         r = self.zoom_session.post(url=url, data=data)
+
+        # Get the scid
+        scid = self.extract_from_js("scid", r.text)
+        token = self.extract_from_js("X-XSRF-TOKEN", r.text)
+
         # Get the XSRF Token
-        pattern = re.search('"X-XSRF-TOKEN".* value:"(.*)"', r.text)
-        if pattern:
+        if token and scid:
             self.zoom_session.headers.update({
-                'X-XSRF-TOKEN': pattern.group(1)
+                'X-XSRF-TOKEN': token
             })
 
-            zoom_json = self.get_zoom_json(1)
+            zoom_json = self.get_zoom_json({'page': 1, 'lti_scid': scid})
             # The first call to zoom returns total and pageSize, get the total pages by dividing
             if zoom_json:
                 total = zoom_json["total"]
                 total_pages = math.ceil(total / zoom_json["pageSize"]) + 1
-            for page_num in range(1, total_pages):
+            for page in range(1, total_pages):
```

Comment:
I like `page_num` a little better, but not a big deal, really.

Reply:
The only reason I changed it is because the parameter on the zoom API call was just *page* I think it's worth being consistent with that.

Reply:
No worries. In my own code, I prefer to distinguish somehow between scalars and objects in the variable name when possible.